autogen/tools/experimental/perplexity/perplexity_search.py

```diff
@@ -0,0 +1,67 @@
+# Copyright (c) 2023 - 2025, AG2ai, Inc., AG2ai open-source projects maintainers and core contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Annotated, Any, Dict, List, Optional
+
+import requests
+
+from autogen.tools import Tool
+
+from .schemas import PerplexityChatCompletionResponse, SearchResponse
+
+
+class PerplexitySearchTool(Tool):
+    def __init__(
+        self,
+        model: str = "sonar",
+        api_key: Optional[str] = None,
+        max_tokens: int = 1000,
+        search_domain_filter: Optional[List[str]] = None,
+    ):
+        self._validate_tool_config(model, api_key, max_tokens, search_domain_filter)
```

Comment:
Can you check if you pass no api_key in that it uses the environment variable and doesn't fail this validation.

E.g. I was thinking:
```suggestion
        self.api_key = api_key or os.getenv("PERPLEXITY_API_KEY")
        self._validate_tool_config(model, self.api_key, max_tokens, search_domain_filter)
```